AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -357,6 +382,14 @@ - (void)performBatchUpdates:(void (^)())updates completion:(void (^)(BOOL))compl
   [self performBatchAnimated:YES updates:updates completion:completion];
 }
 
+- (void)registerSupplementaryNodeOfKind:(NSString *)elementKind
+{
+  ASDisplayNodeAssert(elementKind != nil, @"A kind is needed for supplementary node registration");
+  [_registeredSupplementaryKinds addObject:elementKind];
+  [self registerClass:[UICollectionReusableView class] forSupplementaryViewOfKind:elementKind
```

Comment:
The subclass you're mentioning was created for `UICollectionViewCell` in order to forward the selected and highlighted properties to `ASCellNode`. There isn't really any need to subclass `UICollectionReusableView` at the moment, since this view isn't selectable like `UICollectionViewCell`. Its main job is handling layout change events at the cell level, something that would likely be useful for re-measuring cells when needed. Not something I want to implement for the sake of it right now without a clear goal of how `invalidateLayout` events are being handled from the `ASCollectionView` itself. Definitely a conversation to have in a new issue.